mmpose/datasets/datasets/base/kpt_3d_mview_rgb_img_direct_dataset.py

```diff
@@ -141,3 +139,110 @@ def __getitem__(self, idx):
             results[c] = result
 
         return self.pipeline(results)
+
+    @staticmethod
+    def _sort_and_unique_outputs(outputs, key='sample_id'):
+        """sort outputs and remove the repeated ones."""
+        outputs = sorted(outputs, key=lambda x: x[key])
+        num_outputs = len(outputs)
+        for i in range(num_outputs - 1, 0, -1):
+            if outputs[i][key] == outputs[i - 1][key]:
+                del outputs[i]
+
+        return outputs
+
+    def _get_scale(self, raw_image_size):
+        heatmap_size = self.ann_info['heatmap_size']
+        image_size = self.ann_info['image_size']
+        assert heatmap_size[0][0] / heatmap_size[0][1] \
+               == image_size[0] / image_size[1]
+        w, h = raw_image_size
+        w_resized, h_resized = image_size
+        if w / w_resized < h / h_resized:
+            w_pad = h / h_resized * w_resized
+            h_pad = h
+        else:
+            w_pad = w
+            h_pad = w / w_resized * h_resized
+
+        scale = np.array([w_pad, h_pad], dtype=np.float32)
+
+        return scale
+
+    @staticmethod
+    def rotate_points(points, center, rot_rad):
+        """
+        :param points:  N*2
+        :param center:  2
+        :param rot_rad: scalar
+        :return: N*2
+        """
+        rot_rad = rot_rad * np.pi / 180.0
+        rotate_mat = np.array([[np.cos(rot_rad), -np.sin(rot_rad)],
+                               [np.sin(rot_rad),
+                                np.cos(rot_rad)]])
+        center = center.reshape(2, 1)
+        points = points.T
+        points = rotate_mat.dot(points - center) + center
+
+        return points.T
+
+    @staticmethod
+    def calc_bbox(pose, pose_vis):
+        index = pose_vis[:, 0] > 0
+        bbox = [
+            np.min(pose[index, 0]),
+            np.min(pose[index, 1]),
+            np.max(pose[index, 0]),
+            np.max(pose[index, 1])
+        ]
+
+        return np.array(bbox)
+
+    def _get_cam(self, calib):
+        """Get camera parameters.
+
+        Returns: Camera parameters.
+        """
+        cameras = {}
+        for id, cam in calib.items():
+            sel_cam = {}
+            # note the transpose operation different from from VoxelPose
+            sel_cam['R'] = np.array(cam['R'], dtype=np.float32).T
+            sel_cam['T'] = np.array(cam['T'], dtype=np.float32)
+
+            sel_cam['k'] = np.array(cam['k'], dtype=np.float32)
+            sel_cam['p'] = np.array(cam['p'], dtype=np.float32)
+
+            sel_cam['f'] = [[cam['fx']], [cam['fy']]]
+            sel_cam['c'] = [[cam['cx']], [cam['cy']]]
+
+            cameras[id] = sel_cam
+
+        return cameras
+
+    def _load_files(self):
+        """load related db files."""
+        assert osp.exists(self.cam_file), f'camera calibration file ' \
+            f"{self.cam_file} doesn't exist, please check again"
+        with open(self.cam_file) as cfile:
+            calib = json.load(cfile)
```

Comment:
Can this be done with json_tricks so you don't need to import both `json` and `json_tricks`?

Reply:
Yes, this can be done both with `json_tricks`.